src/game/common/rts/science.cpp

```diff
@@ -61,13 +61,12 @@ void ScienceInfo::Add_Root_Sciences(std::vector<ScienceType> &rootScience) const
 
 void ScienceStore::Reset()
 {
-    for (auto it = m_infoVec.begin(); it != m_infoVec.end(); ++it) {
-        if (*it != nullptr) {
-            *it = reinterpret_cast<ScienceInfo *>((*it)->Delete_Overrides());
```

Comment:
Note: This assignment is obsolete. Delete_Overrides either returns `nullptr` or `this`.